### PR TITLE
Linkerd viz have hardcoded image value in namespace metadata.yml template

### LINKED ISSUE #9480: linkerd-viz have hardcoded image value in namespace-metadata.yml template
### What is the issue?

The image value in namespace-metadata.yml is hardcoded. It's impossible to deploy linkerd-viz with only private registry.

### How can it be reproduced?

Try to deploy the helm chart linkerd-viz without internet access and only use your local private registry.

### Logs, error output, etc

There are no very relevant log in this use case. We just have ImagePullBackOff on the Pod.
```
[root@master ~]# helm install linkerd-viz linkerd-viz-30.3.0.tgz -f viz-values.yml
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /root/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /root/.kube/config
Error: failed post-install: timed out waiting for the condition


[root@master ~]# kubectl get po
NAME                            READY   STATUS             RESTARTS   AGE
metrics-api-7cdbf46688-7qn2b    2/2     Running            0          3m24s
namespace-metadata-gv8fh        0/1     ImagePullBackOff   0          3m22s
prometheus-55459fc6b-wwhhm      2/2     Running            0          3m24s
tap-74b7675b9c-cj8g5            2/2     Running            0          3m24s
tap-injector-5649df9cf6-vs4sq   2/2     Running            0          3m24s
web-748dfbf6cb-pcz25            2/2     Running            0          3m24s

[root@master ~]# kubectl describe po namespace-metadata-gv8fh
Name:         namespace-metadata-gv8fh
Namespace:    default
Priority:     0
Node:         tank5/10.10.0.5
Start Time:   Mon, 26 Sep 2022 11:32:52 +0200
Labels:       app.kubernetes.io/name=namespace-metadata
              app.kubernetes.io/part-of=Linkerd
              app.kubernetes.io/version=stable-2.12.0
              controller-uid=4b661d90-730b-4c12-beb4-be4c5205310c
              job-name=namespace-metadata
Annotations:  linkerd.io/created-by: linkerd/helm stable-2.12.0
              linkerd.io/inject: disabled
Status:       Pending
IP:           192.168.5.165
IPs:
  IP:           192.168.5.165
Controlled By:  Job/namespace-metadata
Containers:
  namespace-metadata:
    Container ID:  
    Image:         curlimages/curl:7.78.0
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
    Args:
      -c
      ops=''
      token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
      ns=$(curl -kfv -H "Authorization: Bearer $token" \
        "https://kubernetes.default.svc/api/v1/namespaces/default")
      
      if ! echo "$ns" | grep -q 'labels'; then
        ops="$ops{\"op\": \"add\",\"path\": \"/metadata/labels\",\"value\": {}},"
      fi
      if ! echo "$ns" | grep -q 'annotations'; then
        ops="$ops{\"op\": \"add\", \"path\": \"/metadata/annotations\", \"value\": {}},"
      fi
      ops="$ops{\"op\": \"add\", \"path\": \"/metadata/labels/linkerd.io~1extension\", \"value\": \"viz\"}"
      
      curl -kfv -XPATCH -H "Content-Type: application/json-patch+json" -H "Authorization: Bearer $token" \
        -d "[$ops]" \
        "https://kubernetes.default.svc/api/v1/namespaces/default?fieldManager=kubectl-label"
      
    State:          Waiting
      Reason:       ImagePullBackOff
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-xtlxt (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  kube-api-access-xtlxt:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  4m31s                  default-scheduler  Successfully assigned default/namespace-metadata-gv8fh to tank5
  Normal   Pulling    2m57s (x4 over 4m31s)  kubelet            Pulling image "curlimages/curl:7.78.0"
  Warning  Failed     2m57s (x4 over 4m30s)  kubelet            Failed to pull image "curlimages/curl:7.78.0": rpc error: code = Unknown desc = Error reading manifest 7.78.0 in registry.regionadmin.svc.kube.local:5000/curlimages/curl: unauthorized: access to the requested resource is not authorized
  Warning  Failed     2m57s (x4 over 4m30s)  kubelet            Error: ErrImagePull
  Warning  Failed     2m46s (x6 over 4m30s)  kubelet            Error: ImagePullBackOff
  Normal   BackOff    2m35s (x7 over 4m30s)  kubelet            Back-off pulling image "curlimages/curl:7.78.0"
```

### output of `linkerd check -o short`

```
[root@master ~]# linkerd viz check
linkerd-viz
-----------
√ linkerd-viz Namespace exists
√ linkerd-viz ClusterRoles exist
√ linkerd-viz ClusterRoleBindings exist
√ tap API server has valid cert
√ tap API server cert is valid for at least 60 days
√ tap API service is running
‼ linkerd-viz pods are injected
    could not find proxy container for namespace-metadata-gv8fh pod
    see https://linkerd.io/2.12/checks/#l5d-viz-pods-injection for hints
√ viz extension pods are running
√ viz extension proxies are healthy
‼ viz extension proxies are up-to-date
    Get "https://versioncheck.linkerd.io/version.json?version=stable-2.12.0&uuid=unknown&source=cli": dial tcp: lookup versioncheck.linkerd.io on 10.10.248.10:53: server misbehaving
    see https://linkerd.io/2.12/checks/#l5d-viz-proxy-cp-version for hints
√ viz extension proxies and cli versions match
√ prometheus is installed and configured correctly
√ can initialize the client
√ viz extension self-check

Status check results are √
```

### Environment

- Kubernetes Version: v1.21.4
- HostOS: RHEL 8.5
- LinkerdVersion: stable-2.12.0

### Possible solution

The solution is simple it is enough to add in the values.yml the possibility of configuring the image. I am already preparing a PR to fix this bug.

### Additional context

_No response_

### Would you like to work on fixing this bug?

yes

### PR DESCRIPTION
**Pull request by bastienbosser**
```
Subject

linkerd-viz have hardcoded image value in namespace-metadata.yml template

Problem

The image value in namespace-metadata.yml is hardcoded. It's impossible to deploy linkerd-viz with only private registry.

Solution

The solution is simple it is enough to add in the values.yml the possibility of configuring the image.

Validation

I build the helm chart and deploy it without internet access (I only use my local registry)

```

Fixes #9480

DCO Sign off
```

Signed-off-by: BOSSER, Bastien <bastien.bosser@atos.net>

```

### CODE REVIEW

File: viz/charts/linkerd-viz/Chart.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ kubeVersion: ">=1.21.0-0"
 name: "linkerd-viz"
 sources:
 - https://github.com/linkerd/linkerd2/
-version: 30.3.3
+version: 30.3.4
```
Comment (adleong) [inline viz/charts/linkerd-viz/Chart.yaml:14]: Our release process is that we only increment this version during the release.  So this should stay at 30.3.3 in this PR.

**Review by adleong — COMMENTED** [review]
Looks great, thanks for working on this.  I have one request about the helm chart version, but otherwise this looks good to me.

**Review by alpeb — COMMENTED** [review]
Thanks @bastienbosser :+1: 
For completeness sake, can I ask you to please commit the same changes to the jaeger and multicluster extensions?

**Comment by bastienbosser** [comment]
@alpeb I'll take care of it.

File: jaeger/charts/linkerd-jaeger/values.yaml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # -- Namespace of the Linkerd core control-plane install
 linkerdNamespace: linkerd
 
+# -- Docker imagePullPolicy for all jaeger components
+defaultImagePullPolicy: IfNotPresent
```
Comment (alpeb) [inline jaeger/charts/linkerd-jaeger/values.yaml:5]: Can you rename this to `imagePullPolicy` to match what we have in the `linkerd-control-plane` chart? (Also for the multicluster chart)

Comment (bastienbosser) [reply jaeger/charts/linkerd-jaeger/values.yaml]: Hi @alpeb, It's done.

File: multicluster/charts/linkerd-multicluster/values.yaml [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+  image:
+    # -- Docker registry for the gateway instance
+    registry: gcr.io
+    # -- Docker image name for the gateway instance
+    name: google_containers/pause
+    # -- Docker image tag for the gateway instance
+    tag: 3.2
+    # -- Pull policy for the gateway instance
+    # @default -- imagePullPolicy
+    pullPolicy: ""
```
Comment (alpeb) [inline multicluster/charts/linkerd-multicluster/values.yaml:13]: I think we should leave this out on this PR, as #9566 is tackling it separately

**Review by alpeb — COMMENTED** [review]
Almost done, just one comment, and please run `bin/helm-docs` to update the charts README files.

File: multicluster/charts/linkerd-multicluster/values.yaml (hunk 3 above)
Comment (bastienbosser) [reply multicluster/charts/linkerd-multicluster/values.yaml:13]: I understand but I find it a pity that it creates a new way to declare the image. It makes the whole chart a bit less coherent.

**Review by alpeb — APPROVED** [review]
LGTM! I took the liberty to merge `main` to have the integration tests pass.

**Review by adleong — APPROVED** [review]

**Comment by kk-bavenkatachalam** [comment]
@alpeb @adleong 
Sorry to bother but could you please let me know if this would be included in the next stable release and when is it scheduled to release?

**Comment by alpeb** [comment]
This is already included in the edge releases, and will be included in 2.13.0. We don't have yet a date to publicly commit to.

**Comment by kk-bavenkatachalam** [comment]
> This is already included in the edge releases, and will be included in 2.13.0. We don't have yet a date to publicly commit to.

Noted, Thanks!

**Pull request merged**